File: mock-platform/python_compat/airline-app/backend/app/models/baggage.py
from app.models import BaseModel, db


class BaggageTracking(BaseModel):
    __tablename__ = "baggage_tracking"

    user_id = db.Column(
        db.Integer, db.ForeignKey("users.id"), nullable=False, index=True
    )
    booking_id = db.Column(db.Integer, db.ForeignKey("bookings.id"))
    flight_number = db.Column(db.String(10), nullable=False)
    flight_time = db.Column(db.DateTime, nullable=False)
    seat_number = db.Column(db.String(5))
    passenger_name = db.Column(db.String(100), nullable=False)
    passenger_phone = db.Column(db.String(20), nullable=False)
    passenger_email = db.Column(db.String(120), nullable=False)
    baggage_description = db.Column(db.Text, nullable=False)
    loss_details = db.Column(db.Text)
    status = db.Column(db.String(20), default="processing", index=True)
    location = db.Column(db.String(200))

    user = db.relationship(
        "User", backref=db.backref("baggage_reports", lazy="dynamic")
    )
    booking = db.relationship(
        "Booking", backref=db.backref("baggage_reports", lazy="dynamic")
    )

    def to_dict(self):
        return {
            "id": self.id,
            "user_id": self.user_id,
            "booking_id": self.booking_id,
            "flight_number": self.flight_number,
            "flight_time": self.flight_time.isoformat() if self.flight_time else None,
            "seat_number": self.seat_number,
            "passenger_name": self.passenger_name,
            "passenger_phone": self.passenger_phone,
            "passenger_email": self.passenger_email,
            "baggage_description": self.baggage_description,
            "loss_details": self.loss_details,
            "status": self.status,
            "location": self.location,
            "created_at": self.created_at.isoformat() if self.created_at else None,
            "updated_at": self.updated_at.isoformat() if self.updated_at else None,
        }

    def __repr__(self):
        return f"<BaggageTracking {self.flight_number} - {self.passenger_name}>"

File: mock-platform/mocks/airline/src/routes/baggage.ts
import type { OpenAPIApp } from "mock-lib";
import type { Database } from "bun:sqlite";
import { ok, err, paginate, parsePageParams } from "../helpers";

export function registerBaggageRoutes(app: OpenAPIApp, db: Database): void {
  // GET /api/baggage
  app.get("/api/baggage", (c) => {
    const query = c.req.query();
    const { page, perPage, offset } = parsePageParams(query.page, query.per_page);
    const status = query.status;

    const userId = c.get("userId")!;
    let sql = "SELECT * FROM baggage_tracking WHERE user_id = ?";
    const params: (number | string)[] = [userId];

    if (status) {
      sql += " AND status = ?";
      params.push(status);
    }

    const countRow = db.query(`SELECT COUNT(*) as total FROM (${sql})`).get(...params) as { total: number };
    const items = db.query(`${sql} ORDER BY created_at DESC LIMIT ? OFFSET ?`).all(...params, perPage, offset) as Record<string, unknown>[];

    return c.json(ok(paginate(items, countRow.total, page, perPage, "baggage_reports")));
  });

  // POST /api/baggage
  app.post("/api/baggage", async (c) => {
    const body = (await c.req.json()) as Record<string, unknown>;
    const flightNumber = String(body.flight_number ?? "");
    const flightTime = String(body.flight_time ?? "");
    const passengerName = String(body.passenger_name ?? "");
    const passengerPhone = String(body.passenger_phone ?? "");
    const passengerEmail = String(body.passenger_email ?? "");
    const baggageDescription = String(body.baggage_description ?? "");
    const seatNumber = body.seat_number ? String(body.seat_number) : null;
    const lossDetails = body.loss_details ? String(body.loss_details) : null;
    const bookingId = body.booking_id ? parseInt(String(body.booking_id), 10) : null;

    if (!flightNumber || !flightTime || !passengerName || !passengerPhone || !passengerEmail || !baggageDescription) {
      return c.json(err("flight_number, flight_time, passenger_name, passenger_phone, passenger_email and baggage_description are required"), 400);
    }

    const userId = c.get("userId")!;
    db.query(
      "INSERT INTO baggage_tracking (user_id, booking_id, flight_number, flight_time, passenger_name, passenger_phone, passenger_email, baggage_description, seat_number, loss_details, status) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, 'processing')"
    ).run(
      userId,
      bookingId,
      flightNumber,
      flightTime,
      passengerName,
      passengerPhone,
      passengerEmail,
      baggageDescription,
      seatNumber,
      lossDetails,
    );

    const reportId = Number((db.query("SELECT last_insert_rowid() as id").get() as { id: number }).id);
    const report = db.query("SELECT * FROM baggage_tracking WHERE id = ?").get(reportId) as Record<string, unknown>;
    return c.json(ok(report, "Baggage report submitted successfully"), 201);
  });

  // GET /api/baggage/:report_id
  app.get("/api/baggage/:report_id", (c) => {
    const id = parseInt(c.req.param("report_id"), 10);
    const userId = c.get("userId")!;
    const report = db.query("SELECT * FROM baggage_tracking WHERE id = ? AND user_id = ?").get(id, userId) as Record<string, unknown> | null;
    if (!report) return c.json(err("Baggage report not found"), 404);
    return c.json(ok(report));
  });
}
